File: 프로그래머스/unrated/181867. x 사이의 개수/x 사이의 개수.cpp
#include <string>
#include <vector>
#include <sstream>
#include <iostream>
using namespace std;

vector<int> solution(string myString) {
    vector<int> answer;
    istringstream ss(myString);
    string buffer = "";
    while(getline(ss, buffer, 'x')){
        answer.push_back(buffer.size());
    }
    if(myString[myString.size()-1] == 'x') answer.push_back(buffer.size());
    return answer;
}

File: 프로그래머스/unrated/181867. x 사이의 개수/x 사이의 개수.java
import java.util.*;

class Solution {
    public int[] solution(String myString) {
        List<Integer> answer = new ArrayList<Integer>();
        String[] rs = myString.split("x");
        for(String s : rs){
            answer.add(s.length());
        }
        if(myString.charAt(myString.length()-1) == 'x')
            answer.add(0);
        return answer.stream().mapToInt(i->i).toArray();
    }
}